### PR TITLE
refactor: Change lib v2 code to TS

### PR DESCRIPTION
**Pull request by yusuf-musleh**
This PR is temp for review, should be replaced with https://github.com/openedx/frontend-app-course-authoring/pull/1124 once base PR is merged.

### CODE REVIEW

File: src/studio-home/tabs-section/libraries-v2-tab/index.tsx [hunk 1]
```diff
@@ -98,7 +98,7 @@ const LibrariesV2Tab = ({
         </div>
 
         { hasV2Libraries
-          ? data.results.map(({
+          ? data?.results.map(({
```
Comment (bradenmacdonald) [inline src/studio-home/tabs-section/libraries-v2-tab/index.tsx]: react-query guarantees that only one of `data`, `isLoading`, or `isError` will be truthy at any given time, so you can either change this to `data!.results` or refactor the code like this:

```ts
if (isLoading) {
  // Display loading state
}
if (isError || data === undefind) {
  // Display an error message. The `|| data === undefined` is only needed for TypeScript to understand the flow.
}

// Render the normal component. Here TypeScript now knows that `data` is not undefined
```

**Review by pomegranited — APPROVED** [review]
:+1: This works as expected, thank you @yusuf-musleh 

- [x] I tested this on my tutor dev stack using the instructions from https://github.com/openedx/frontend-app-course-authoring/pull/1050
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] User-facing strings are extracted for translation

File: src/studio-home/tabs-section/libraries-v2-tab/index.tsx (hunk 1 above)
Comment (yusuf-musleh) [reply src/studio-home/tabs-section/libraries-v2-tab/index.tsx]: I see, I've updated it to use the `data!` and made sure that the to check for `!isLoaded` in the `hasV2Libraries` which is used in a bunch of places. (`isError` is already checked at the beginning and returns): [a72713e](https://github.com/open-craft/frontend-app-course-authoring/pull/43/commits/a72713e34decb07215a236f85ff70d5d6ff9f21e)

**Comment by yusuf-musleh** [comment]
@pomegranited Thanks for the review!

> Note there is a bug that linting for .ts[x] files is not happening properly; I'm fixing it in https://github.com/openedx/frontend-app-course-authoring/pull/1129 but please manually make [this change](https://github.com/open-craft/frontend-app-course-authoring/commit/9f00846e54a8b865878dc0a29189cbca2bc9e1e4) and run npm run lint yourself to make sure this PR is good too.

@bradenmacdonald I made sure to run the linter and there are no issues with the file changes in this PR.

**Pull request merged**